### DISCUSSION TITLE
Using 2.0 style ORM in my v1.4 version of sqlalchemy

### DISCUSSION THREAD

**Discussion by taigit** [sqlalchemy/sqlalchemy#8658] [Usage Questions] [answered]
Q: Can I use the 2.0 style ORM in my v1.4 version of sqlalchemy?

2.0 style ORM  = https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-orm-usage

A:  you can use 2.0 style in 1.4, certainly, for select(), update(), and delete() statements.
https://github.com/sqlalchemy/sqlalchemy/discussions/8157#discussioncomment-3879080

_________________________

That's great news!  Now, I'm trying to understand the reasoning. Is this reasoning correct:

if I am using sqlalchemy 1.4 and I do the 5 migration steps in:
https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-to-2-0-step-five-use-the-future-flag-on-session

then, I'm actually running sqlalchemy 2.0 and can use the 2.0 style ORM and can use the 2.0 documentation (except for the new 2.0 features) even though I'm running sqlalchemy 1.4

Is that the correct reasoning?

And, in fact, at this point my code can run either in v1.4 or v2.0 of sqlalchemy.

**Comment by zzzeek** [comment, marked as answer]
> That's great news! Now, I'm trying to understand the reasoning. Is this reasoning correct:
> 
> if I am using sqlalchemy 1.4 and I do the 5 migration steps in: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-to-2-0-step-five-use-the-future-flag-on-session
> 
> then, I'm actually running sqlalchemy 2.0 and can use the 2.0 style ORM and can use the 2.0 documentation (except for the new 2.0 features) even though I'm running sqlalchemy 1.4
> 
> Is that the correct reasoning?
> 
> And, in fact, at this point my code can run either in v1.4 or v2.0 of sqlalchemy.

that's correct!
